src/main/java/Fundamentals/oop/BankAccount.java

```diff
@@ -0,0 +1,63 @@
+package Fundamentals.oop;
+
+public class BankAccount {
+    private String type;
+    private double balance;
+    private String currency;
+    private String accountType;
+
+    public BankAccount(String type, String currency) {
+        this.type = type;
+        this.currency = currency;
+        this.balance = 0;
+    }
+
+    public BankAccount() {
+        this.type = "SAVING";
+        this.currency = "USD";
+        this.balance = 0;
+    }
+
+    public void AddMoney(double newMoney) {
+        if (this.accountType == "SPENDING") {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [ES: Comparison of String objects using == or != (ES_COMPARING_STRINGS_WITH_EQ)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#es-comparison-of-string-objects-using-or-es-comparing-strings-with-eq).
This code compares java.lang.String objects for reference equality using the == or != operators. Unless both strings are either constants in a source file, or have been interned using the String.intern() method, the same string value may be represented by two different String objects. Consider using the equals(Object) method instead.